tools/src/linter/SchemasValidator.ts

```diff
@@ -0,0 +1,100 @@
+import AJV from 'ajv'
+import addFormats from 'ajv-formats'
+import OpenApiMerger from '../merger/OpenApiMerger'
+import { type ValidationError } from '../types'
+
+export default class SchemasValidator {
+  root_folder: string
+  spec: Record<string, any> = {}
+  ajv: AJV
+
+  constructor (root_folder: string) {
+    this.root_folder = root_folder
+    this.ajv = new AJV()
+    addFormats(this.ajv)
+  }
+
+  validate (): ValidationError[] {
+    this.spec = new OpenApiMerger(this.root_folder, true).merge().components as Record<string, any>
```

Comment:
What's the reasoning for needing to run against the merged output? 

I naively would think something like below would work:
```typescript
import AJV from 'ajv'
import addFormats from 'ajv-formats'
import fs from 'fs'
import { type OpenAPIV3 } from 'openapi-types'
import { type ValidationError } from '../types'
import { SchemaVisitor } from './utils/SpecificationVisitor'
import { is_ref, type MaybeRef, SpecificationContext } from './utils'
import { read_yaml } from '../../helpers'

export default class SchemasValidator {
  root_folder: string
  ajv: AJV

  constructor (root_folder: string) {
    this.root_folder = root_folder
    this.ajv = new AJV()
    addFormats(this.ajv)
  }

  validate (): ValidationError[] {
    const errors: ValidationError[] = []

    const visitor = new SchemaVisitor((ctx, schema) => {
      this.#validate_schema(ctx, schema, errors)
    })

    const list_folder = (subdir: string): string[] =>
      fs.readdirSync(`${this.root_folder}/${subdir}`)
        .filter((f) => f.endsWith('.yaml'))
        .map((f) => `${subdir}/${f}`)

    const files: string[] = [
      '_global_parameters.yaml',
      ...list_folder('namespaces'),
      ...list_folder('schemas')
    ].sort()

    for (const file of files) {
      visitor.visit_specification(new SpecificationContext(file), read_yaml(`${this.root_folder}/${file}`) as OpenAPIV3.Document)
    }

    return errors
  }

  #validate_schema (ctx: SpecificationContext, schema: MaybeRef<OpenAPIV3.SchemaObject>, errors: ValidationError[]): void {
    if (is_ref(schema)) return

    try {
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      this.ajv.validateSchema(schema, true)
    } catch (err) {
      errors.push(ctx.error((err as Error).message))
    }
  }
}
```

Reply:
This was my original strat at first. But say if you want to validate:
```yaml
# file: schemas/_common.yaml

type: array
items:
  $ref: #/components/schemas/Index
```

Then ajv must be made aware of they key `#/components/schemas/Index` beforehand with
```ts
ajv.addSchema(index_schema, '#/components/schemas/Index')
```

This gets complicated when the `Index` schema is referenced in different files:
- The key is `#/components/schemas/Index` in the `schemas/_common.yaml` file itself
- The key is `_common.yaml#/components/schemas/Index` in a namespace file
- The key is `../schemas/_common.yaml#/components/schemas/Index` in a namespace file

Meaning that whenever you want to validate an individual file, you will have to load up all schemas files, and add the referenced schemas using `ajv.addSchema` accordingly. It's too complicated and not as efficient as loading up the single-file spec.